src/server.rs

```diff
@@ -281,14 +279,81 @@ impl<State: Send + Sync + 'static> Server<State> {
 
     /// Asynchronously serve the app at the given address.
     #[cfg(feature = "h1-server")]
-    pub async fn listen(self, addr: impl ToSocketAddrs) -> std::io::Result<()> {
+    pub async fn listen(self, addr: impl ToSocketAddrs) -> io::Result<()> {
         let listener = async_std::net::TcpListener::bind(addr).await?;
 
         let addr = format!("http://{}", listener.local_addr()?);
         log::info!("Server is listening on: {}", addr);
-        let mut server = http_service_h1::Server::new(addr, listener.incoming(), self);
+        let mut incoming = listener.incoming();
```

Comment:
Sorry to butt into a PR, but this could be a convenient opportunity to consider a separate method that takes a `TcpListener` (or `impl Into<TcpListener>`), even if it's private for now.

I have an agenda as the author of #485 – a side-effect of this change would be making it more difficult to set up Tide with auto-reloading. Specifically, this change would make it infeasible to 'reimplement' `tide::Server::listen` with a pre-existing `TcpListener`.

Apologies if this is redundant with other planned work.

Reply:
There's work in the direction of supporting any transports that's AsyncRead+AsyncWrite+Clone+Send+Sync+'static. Once these changes hit, regardless of which convenience functions exit on Server, it will likely be able to directly implement your own listen loop for any transport, along the lines of https://gist.github.com/yoshuawuyts/e02cc9626aea265a97c181e6cc3baebe

Reply:
Awesome, I did think #486 would supersede my specific concerns.

Reply:
And in the near future, it'll look something like this https://gist.github.com/jbr/183aaeb835aa30c3a5f94ee91cdc473b — very similar but slightly simpler, only two args to async_h1::accept